### PR TITLE
[Accuracy diff No.125] Fix accuracy diff for paddle.nn.functional.multi_margin_loss API

### PR DESCRIPTION
**Pull request by co63oc**
### PR Category

Operator Mechanism

### PR Types

Bug fixes

### Description

https://www.paddlepaddle.org.cn/documentation/docs/zh/api/paddle/nn/functional/multi_margin_loss_cn.html

image
multi_margin_loss  weight 应加到 pow计算后，因为有在0和运算值中取较大的值计算，如果weight有负值，计算会有误差

同样修改单测的计算

PaddleAPITest 测试通过，如果测试错误时会随机出现，需要重复运行几次
image

### CODE REVIEW

**Comment by co63oc** [comment]
@lshpku CI已完成需要review

**Review by lshpku — APPROVED** [review]
